### DISCUSSION TITLE
increase time on npx hardat node

### DISCUSSION THREAD

**Discussion by kypanz** [NomicFoundation/hardhat#4671] [General]
Hello guys, i have a question, is possible to increase the time when i have a node running with `npx hardhat node` ?, i wanna test some functions that need to increment the time, i know i can do it in test cases, but actually i am doing some tests on my backend api and when the transaction is done i wanna test if the websocket is emited to the frontend

sumary : any knows a way to increase the time when are runned with `npx hardhat node` ?

thanks for read and have a nice week :)

**Comment by fvictorio** [comment]
What do you mean by "increase the time"?

What you can do is to configure [interval mining](https://hardhat.org/hardhat-network/docs/explanation/mining-modes), so that blocks are mined every N seconds.

If what you want is to delay the mining/execution of each transaction by a given value, that's not easy to do, but you could [extend the provider](https://hardhat.org/hardhat-runner/docs/advanced/building-plugins#extending-the-hardhat-provider) to do it (this is an advanced feature though)

**Reply by kypanz** [reply]
Hello, thanks for answer and sorry about my english, when i say "increase the time" i mean increase the days on testing

- take a look of this docs : https://hardhat.org/hardhat-network-helpers/docs/reference#increase(amountinseconds)

there has a function called : `await helpers.time.increase(3600);`, this function increase the time by one hour

so this works fine if i run `npx hardhat test`, but currently i am trying to testing some things connecting my `backend + smart contracts`, in this case i am not using `npx hardhat test` i am using `npx hardhat node` to start a node to test how works all the things connected

so my question is how exactly i can "increase the time" like `await helpers.time.increase(3600);`, but using `npx hardhat node`

thanks for read and have a nice weekend

**Reply by fvictorio** [reply]
Ah, I see.

When you start a node with `npx hardhat node`, you can connect to it through `localhost:8545`. This is what you are probably already doing in your dapp.

You can also connect to it by using the `localhost` network (which comes pre-configured in hardhat), so you could have a script that just does:

```js
const helpers = require("@nomicfoundation/hardhat-network-helpers");

async function main() {
  await helpers.time.increase(3600);
}

main();
```

and then run it with `npx hardhat run --network localhost scripts/my-script.js`. Of course, you can do whatever you want within that script.

**Reply by kypanz** [reply]
lol how i not think that before, i am 99% sure that can works, i gonna try it today and tell you if works, thanks you very much and have a nice weekend

**Reply by kypanz** [reply]
Hello again, i think is solved

The output in `npx hardhat node` are : 
```shell
web3_clientVersion
eth_getBlockByNumber
evm_setNextBlockTimestamp
hardhat_mine
  Mined empty block #1 with base fee 875000000

eth_getBlockByNumber

```

I am not sure, but in theory thats works because in deep when i use `npx hardhat test` insinde that instruction runs at the same like `npx hardhat node`

I need to test writing some test code to see if works fine, but i think is solved, i used the example code that you mentioned

Thanks for the answers and the patience, have a nice week :heart: 

![image](https://github.com/NomicFoundation/hardhat/assets/37570367/2829ff0d-70a5-41b8-aef9-f969162c7f2a)
